src/swap.ts

```diff
@@ -0,0 +1,284 @@
+import { purry } from './purry';
+import { isString } from './isString';
+import { isArray } from './isArray';
+import { IterableContainer } from './_types';
+import { Joined } from './join';
+
+/**
+ * @link https://github.com/sindresorhus/type-fest/blob/main/source/is-equal.d.ts
+ */
+type isEqual<A, B> = (<G>() => G extends A ? 1 : 2) extends <G>() => G extends B
+  ? 1
+  : 2
+  ? true
+  : false;
+
+type Difference<A extends number, B extends number> = TupleOfLength<A> extends [
+  ...infer U,
+  ...TupleOfLength<B>
+]
+  ? U['length']
+  : never;
+
+type isLessThan<A extends number, B extends number> = isEqual<A, B> extends true
+  ? false
+  : 0 extends A
+  ? true
+  : 0 extends B
+  ? false
+  : isLessThan<Difference<A, 1>, Difference<B, 1>>;
+
+type TupleOfLength<
+  L extends number,
+  T extends ReadonlyArray<unknown> = []
+> = T['length'] extends L ? T : TupleOfLength<L, [...T, unknown]>;
+
+type IsNonNegativeInteger<T extends number> = number extends T
+  ? false
+  : `${T}` extends `-${string}`
+  ? false
+  : true;
+
+type Absolute<T extends number> = `${T}` extends `-${infer R}`
+  ? R extends `${infer N extends number}`
+    ? N
+    : never
+  : T;
+
+type CharactersTuple<T extends string> = string extends T
+  ? Array<string>
+  : T extends `${infer C}${infer R}`
+  ? [C, ...CharactersTuple<R>]
+  : [];
+
+type ObjectSwap<T, K1 extends keyof T, K2 extends keyof T> = {
+  [K in keyof T]: T[K1 extends K ? K2 : K2 extends K ? K1 : K];
+};
+
+type SwapArrayInternal<
+  T extends IterableContainer,
+  Index1 extends number,
+  Index2 extends number,
+  Position extends IterableContainer = [],
+  Original extends ReadonlyArray<unknown> = T
```

Comment:
This is the other way around
```suggestion
  Position extends ReadonlyArray<unknown> = [],
  Original extends IterableContainer = T
```